test_scripts/API/ButtonSubscription/005_Subscribe_button_SDL_send_UnsubscribeButton_after_Subscribe_SUCCESS_timeout.lua

```diff
@@ -0,0 +1,56 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0192-button_subscription_response_from_hmi.md
+-- User story: TBD
+-- Use case: TBD
+--
+-- Requirement summary: TBD
+--
+-- Description:
+-- In case:
+-- 1) SDL return GENERIC_ERROR to Mobile App after default timeout.
+-- 2) HMI sending Subscribe SUCCESS to SDL.
+-- SDL does:
+-- 1) Send UnsubscribeButton to HMI.
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/ButtonSubscription/common_buttonSubscription')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local variable ]]
+local buttonName = "OK"
+local errorCode = "GENERIC_ERROR"
+
+--[[ Local function ]]
+function rpcGenericError(pAppId, pRpc, pButtonName)
+    local cid = common.getMobileSession(pAppId):SendRPC(pRpc, { buttonName = pButtonName })
+    appIdVariable = common.getHMIAppId(pAppId)
+    EXPECT_HMICALL("Buttons." .. pRpc,{ appID = appIdVariable, buttonName = pButtonName })
+    :Do(function()
+        -- HMI did not response
+    end)
+    common.getMobileSession(pAppId):ExpectResponse(cid, { success = false, resultCode = errorCode })
+    :Do(function()
+        common.getHMIConnection():SendResponse( appIdVariable, "Buttons.SubscribeButton", "SUCCESS", { })    
+    end)
+    EXPECT_HMICALL("Buttons.UnsubscribeButton",{ appID = appIdVariable, buttonName = pButtonName })
+    :Do(function(_, data)
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+    end)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("App registration", common.registerAppWOPTU)
+runner.Step("App activation", common.activateApp)
+
+runner.Title("Test")
+runner.Step("Failure Subscribe on " .. buttonName, rpcGenericError, { 1, "SubscribeButton", buttonName, errorCode })
```

Comment:
i think that better to rename step to "Button unsubscription in case HMI responds after request timeout" or something like this, the main idea of the case is not the Failure Subscribe